zrcs/zrc-2.md

```diff
@@ -0,0 +1,372 @@
+| eventName | `TansferFromSuccess` | Approval is successful.     | `from`: `ByStr20`, `to`: `ByStr20`, and `amount`: `Uint128`.                                                                                               |
+| eventName | `Error`          | Approval is not successful. | - emit `CodeNotAuthorised` if the transition is called by a user who is not authorized to approve. <br>**NOTE:** Only either the `tokenOwner` or approved `operator`(s) are allowed to call this transition. |
+
+#### 13. Allowance
+
+```ocaml
+(* @dev: Returns the number of tokens spender is allowed to spend on behalf of owner. *)
+(* param tokenHolder:  Address of a token holder.                                     *)
+(* param spender:      Address to be set as a spender.                                *)
+transition Allowance(tokenHolder: ByStr20, spender: ByStr20)
+```
+
+|        | Name      | Type      | Description                                    |
+| ------ | --------- | --------- | ---------------------------------------------- |
+| @param | `tokenHolder` | `ByStr20` | Address of a token owner.                               |
+| @param | `spender`     | `ByStr20` | Address to be set as a spender for a given token owner. |
+
+|           | Name             | Description                 | Event Parameters                                                                                                                                                                                                                                              |
+| --------- | ---------------- | --------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| eventName | `AllowanceSuccess` | Allowing is successful.     | `tokenHolder`: `ByStr20`, and `spender`: `ByStr20`.                                                                                               |
+| eventName | `Error`          | Allowing is not successful. | TBA. |
+
+
+#### 14. Approve
+
+```ocaml
+(* @dev: Sets amount as the allowance of spender over the caller’s tokens.  *)
+(* There can only be one approved spender per token at a given time         *)
+(* param tokenHolder:  Address of a token holder.                           *)
+(* param spender:      Address to be set as a spender.                      *)
+(* param amount:       Number of tokens to be approved for a given spender. *)
+transition Approve(tokenHolder: ByStr20, spender: ByStr20, amount: Uint128)
+```
+
+|        | Name      | Type      | Description                                          |
+| ------ | --------- | --------- | ---------------------------------------------------- |
+| @param | `amount`  | `ByStr20` | Address of a token owner.                            |  
```

Comment:
`amount` here must be replaced with `tokenOwner`

Reply:
Got it. I've replaced `tokenHolder` with `tokenOwner` throughout.